react-validated-input-component/src/index.jsx

```diff
@@ -0,0 +1,37 @@
+import React, { useState } from 'react';
+import reactDOM from 'react-dom/client';
+const container = document.querySelector('#root');
+const root = reactDOM.createRoot(container);
+
+function PasswordField() {
+  const [password, setPassword] = useState('');
+  const isGoodEnough = password.length >= 8;
+
+  const passwordHelper = () => {
+    if (password.length === 0) {
+      return <span className='red-text'>a password is required</span>;
+    } if (password.length <= 7) {
+      return <span className='red-text'>password too short</span>;
+    } if (password.length > 7) {
+      return '';
+    }
+  };
```

Comment:
Good. I have one complaint, though, and that is that `isGoodEnough` and `passwordHelper` can get out of sync. You are fortunate that your check is only checking the length. What if you add the other checks, like making sure the password has the correct characters?

It would be best to do the check once, perhaps in `passwordHelper` and then check that return value to see if you should display the check mark.

Reply:
ty for the tips, definitely will be switching things around to do the challenge parts! just wanted to make sure it worked enough for submit then I can fart around